bin/find_mono_energy

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+
+
+"""Compute monochromatic energy based on spectral model
+(incl responses) and band"""
+
+import sys
+
+import numpy as np
+from sherpa.astro.ui import *       # pylint: disable=wildcard-import
+from paramio import pset, paramopen, paramclose
+import ciao_contrib.logger_wrapper as lw
+
+TOOLNAME = "find_mono_energy"
+__REVISION__ = "28 June 2021"
+
+DATASET_ID = "myid"
+
+# pylint: disable=undefined-variable
+
+
+def setup_dataspace(arf, rmf):
+    """Setup the sherpa dataspace based on RMF grid and associate
+    ARF and RMF with that dataset.
+    """
+
+    _rmf = unpack_rmf(rmf)
+    chan_lo = _rmf.offset
+    chan_hi = _rmf.detchans
+
+    _arf = unpack_arf(arf)
+
+    dataspace1d(chan_lo, chan_hi, id=DATASET_ID, dstype=DataPHA)
+    set_arf(DATASET_ID, _arf)
+    set_rmf(DATASET_ID, _rmf)
+
+
+def setup_model(model, model_params):
+    "Set the source model"
+
+    # pylint: disable=exec-used
+
+    set_model(DATASET_ID, model)
+    exec(model_params)
```

Comment:
Perhaps we should have this in a `try/except` block so we can say something like (but tested)

````
try:
    exec(model_params)
except Exception as e:
    raise ValueError(f'Error in paramvals={model_params}]\n{e}')
```